### PR TITLE
Update 1-Two-Sum.java improved code readability

### PR DESCRIPTION
**Pull request by ColstonBod-oy**
Modified the way the variables are used and named as I find them a bit confusing at first glance.

### CODE REVIEW

File: java/1-Two-Sum.java [hunk 1]
```diff
@@ -3,16 +3,16 @@ public int[] twoSum(int[] nums, int target) {
...
+            int diff = target - nums[i];
             
-            if (prevMap.containsKey(nums[i])) {
-                return new int[]{prevMap.get(num), i};
+            if (prevMap.containsKey(diffNum)) {
+                return new int[]{prevMap.get(diffNum), i};
             }
             
-            prevMap.put(target - num, i);
+            prevMap.put(diff, i);
```
Comment (mitchellirvin) [inline java/1-Two-Sum.java:13]: this line change makes sense!

**Review by mitchellirvin — CHANGES_REQUESTED** [review]
hi @ColstonBod-oy ! thank you for the contribution!

i'm not sure about the name `diffNum`. it implies "difference number", but that's not actually what it is (it's not the result of subtraction operation). it is the "current number in a loop over a collection of numbers". also, `diffNum` now clashes with `diff`, and it's not clear looking at them side by side what the difference is.

it's pretty standard to use the convention: `int type = types[i];` when capturing the current element in a loop, so i think the existing code of `int num = nums[i];` makes sense. you usually only see a more specific name, maybe `currentNum` or the like when there is a "next" or "previous" to track as well. 

i think the change on line 13 is a positive one, but in general I don't think the other changes improve the readability.

File: java/1-Two-Sum.java (hunk 1 above)
Comment (ColstonBod-oy) [reply java/1-Two-Sum.java:13]: @mitchellirvin I see, I was kinda hesitant of the new variable name as well, thanks for clarifying!

**Comment by ColstonBod-oy** [comment]
Changed the code on line 13 to use the variable ```diff``` instead.

**Pull request closed without merging**